extensions/intents-3.3.md

```diff
@@ -0,0 +1,93 @@
+---
+title: IRCv3.3 Message Intents framework
+layout: spec
+copyrights:
+  -
+    name: "William Pitcock"
+    period: "2012"
+    email: "nenolod@dereferenced.org"
+  -
+    name: "Kiyoshi Aman"
+    period: 2016
+    email: "kiyoshi.aman@gmail.com"
+---
+This specification concerns adding an optional out-of-band message tag to messages
+using the message tagging framework.  Message intent tags should be considered an
+initial step in replacing CTCP, as well as providing strong replacements for several
+interpretations of `NOTICE's` intent.
+
+## Background
+
+### CTCP intent encapsulation
+
+CTCP provides a way of encapsulating messages with intent information.  This is done
+by encapsulating the message body inside 0x1 bytes, with the first word in the message
+body being a keyword which identifies the intent of the message, such as "ACTION" or
+"DCC".
+
+One purpose of this specification is to obsolete this encoding method using the message
+tagging framework included in IRCv3.
+
+### `NOTICE` meanings
+
+A lot of clients, and some servers, assign various meanings to `NOTICE`, in addition to
+the RFC 1459 meaning of "please do not reply to this message". This specification provides
+ways to apply intentions to any message, not just `NOTICE`.
+
+## The `intents` capability
+
+A client MUST request the `intents` capability in order to receive intent tags.  IRCd
+software MAY represent intent tagged messages in the legacy encapsulation format described
+above for clients which did not request the `intents` capability.
+
+## The `intent` message tag
+
+The `intent` tag carries the sender's intentions on how the message should be interpreted.
+The following intentions SHOULD be supported:
+
+| Intention   | CTCP? | Additional tag      | Meaning                                        |
+|-------------|-------|---------------------|------------------------------------------------|
+| `action`    | Yes.  | N/A                 | This message is an 'action'.                   |
+| `announce`  | No.   | N/A                 | Mark this message as an announcement.          |
+| `error`     | No.   | N/A                 | This message is an 'error' notification.       |
+| `log`       | No.   | `loglevel=<string>` | This message contains logging information.     |
```

Comment:
As I've said before, I don't think they're significant enough to deserve their own specification, particularly since they fit within the umbrella of client-side intended behaviour.